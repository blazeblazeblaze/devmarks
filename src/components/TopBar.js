import React from "react";
import { Link } from "react-router-dom";

const TopBar = () => (
  <nav className="navbar has-shadow is-spaced">
    <div className="container">
      <div className="navbar-brand">
        <Link className="navbar-item" to="/">
          <div className="icon has-text-danger">
            <i className="far fa-bookmark fa-lg" />
          </div>
        </Link>
        <div
          className="navbar-burger burger"
          data-target="navbarExampleTransparentExample"
        >
          <span />
          <span />
          <span />
        </div>
      </div>

      <div className="navbar-menu">
        <div className="navbar-start">
          <Link className="navbar-item is-active" to="/">
            <span className="icon">
              <i className="fas fa-code" />
            </span>
            <span>Programming</span>
          </Link>
        </div>

        <div className="navbar-end">
          <div className="navbar-item">
            <div className="field is-grouped">
              <p className="control">
                <a
                  className="bd-tw-button is-medium button"
                  target="_blank"
                  rel="noopener noreferrer"
                  href="http://github.com/blazejhadzik/devmarks"
                >
                  <span className="icon">
                    <i className="fab fa-github" />
                  </span>
                  <span>Make it your own</span>
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
);

export default TopBar;
