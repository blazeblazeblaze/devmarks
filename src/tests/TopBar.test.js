import React from "react";
import TopBar from "../components/TopBar";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<TopBar />", () => {
  it("matches snapshot", () => {
    const tree = shallow(<TopBar />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
