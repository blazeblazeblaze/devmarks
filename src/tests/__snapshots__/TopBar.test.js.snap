// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> matches snapshot 1`] = `
<nav
  className="navbar has-shadow is-spaced"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <Link
        className="navbar-item"
        replace={false}
        to="/"
      >
        <div
          className="icon has-text-danger"
        >
          <i
            className="far fa-bookmark fa-lg"
          />
        </div>
      </Link>
      <div
        className="navbar-burger burger"
        data-target="navbarExampleTransparentExample"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-start"
      >
        <Link
          className="navbar-item is-active"
          replace={false}
          to="/"
        >
          <span
            className="icon"
          >
            <i
              className="fas fa-code"
            />
          </span>
          <span>
            Programming
          </span>
        </Link>
      </div>
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <div
            className="field is-grouped"
          >
            <p
              className="control"
            >
              <a
                className="bd-tw-button is-medium button"
                href="http://github.com/blazejhadzik/devmarks"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  className="icon"
                >
                  <i
                    className="fab fa-github"
                  />
                </span>
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
